### ISSUE TITLE
PHP Less Compiler doesn't properly compile nested selectors within conditions

### ISSUE THREAD

**Issue by craft-tyler** [magento/magento2#12864]
### Preconditions

1. Magento Cloud
2. Production Mode
3. Magento2 EE 2.2.0
4. Using svg-sprite grunt module - https://github.com/jkphl/grunt-svg-sprite

### Steps to reproduce

The issue we are running into is within the PHP less compiler.  When nesting selectors inside of conditional statements the php less compile ignores the conditioanls and doesn't render properly.  See example:

```
.svg-view() {
  background: url("../images/sprites.view.svg") no-repeat;
}

.svg-view {
.svg-view();
}

.svg-view-slider-left {
    background-position: 0 0;
}

.svg-view-slider-left-dims {
    width: 43px;
    height: 43px;
}

.svg-icon(@fileName) when(isstring(@fileName)) {
.svg-view();

  & when (@fileName = "slider-left") {
      .svg-view-slider-left;
      .svg-view-slider-left-dims;
  }
}

body {
    .svg-icon('slider-left');
}
```

### Expected result

Expected CSS Output: 
See [codepen](https://codepen.io/tcraft/pen/JMRzJj) with working example using LESS js compiler
(view compiled code)

```
body {
  background: url("../images/sprites.view.svg") no-repeat;
  background-position: 0 0;
  width: 43px;
  height: 43px;
}
```

### Actual result

Actual results compiled from Less PHP Compiler:

```
body {
  background: url("http://images/sprites.view.svg") no-repeat;
}
```

Notice the PHP Compiler ignores the `.svg-view-slider-left;' and '.svg-view-slider-left-dims;` selectors while nested in the `& when (@fileName = "slider-left") {` conditional statement which is native LESS behavior.

This is native LESS behavior which doesn't work on production environments due to the choice of leveraging the php LESS compiler.

**Comment by magento-engcom-team** [comment]
@craft-tyler thank you for your report.
Could you please let us know if the issue reproduces also in version CE without  "svg-sprite grunt module" or any other third-party extension?

**Comment by craft-tyler** [comment]
Yes, this is replicable within a Native Magento instance on CE.  Although, this is more an issue with the php less compiler module in production.  This issue will occur in custom less functions if there are conditional statements in them...

**Comment by magento-engcom-team** [comment]
@craft-tyler, thank you for your report.
We've created internal ticket(s) MAGETWO-86178 to track progress on the issue.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by ghost** [comment]
Hi @craft-tyler thank you for you report, this issue has already fixed in 2.3-develop branch and will be available on 2.3.1 release.
